common-npm-packages/webdeployment-common/Tests/L0CopyDirectory.ts

```diff
@@ -1,59 +1,91 @@
-var mockery = require('mockery');
-mockery.enable({
-    useCleanCache: true,
-    warnOnReplace: false,
-    warnOnUnregistered: false
-});
-
-var fileList = ["C:/vinca/path", "C:/vinca/path/myfile.txt",
-                "C:/vinca/path/New Folder", "C:/vinca/path/New Folder/Another New Folder",
-                "C:/vinca/New Folder/anotherfile.py", "C:/vinca/New Folder/Another New Folder/mynewfile.txt"];
-
-var mkdirPCount = 0;
-var cpfilesCount = 0;
-mockery.registerMock('azure-pipelines-task-lib/task', {
-    exist: function (path) {
-        console.log("exist : " + path);
-    },
-    find: function (path) {
-        console.log("find : " + path);
-        return fileList;
-    },
-    mkdirP: function (path) {
-        mkdirPCount += 1;
-        console.log("mkdirp : " + path);
-    },
-    cp: function (source, dest, options, continueOnError) {
-        if(fileList.indexOf(source)!= -1) {
-            cpfilesCount += 1;
-            console.log('cp ' + source + ' to ' + dest);
-        }
-    },
-    stats: function (path) {
-        return {
-            isDirectory: function() {
-                if(path.endsWith('.py') || path.endsWith('.txt')) {
-                    return false;
+import * as assert from "assert";
+import * as mockery from "mockery";
+import * as path from "path";
+
+export function runCopyDirectoryTests(): void {
+    const fileList: string[] = [];
+    let mkdirPCount: number;
+    let cpfilesCount: number;
+
+    before(() => {
+
+        const taskLibMock = {
+            exist: function (path: string): boolean {
+                console.log("exist : " + path);
+                return fileList.indexOf(path) !== -1;
+            },
+            find: function (path: string): string[] {
+                console.log("find : " + path);
+                return fileList.filter(f => f.startsWith(path));
+            },
+            mkdirP: function (path: string): void {
+                if (fileList.indexOf(path) !== -1) {
+                    return;
                 }
-                return true;
+                
+                mkdirPCount++;
+                fileList.push(path);
+                console.log("mkdirp : " + path);
+            },
+            cp: function (source: string, dest: string, _options: any, _continueOnError: boolean): void {
+                if (fileList.indexOf(source) === -1) {
+                    return;
+                }
+                if (fileList.indexOf(dest) !== -1) {
+                    return;
+                }
+                cpfilesCount++;
+                fileList.push(dest);
+                console.log('cp ' + source + ' to ' + dest);
+            },
+            stats: function (path: string): any {
+                return {
+                    isDirectory: function (): boolean {
+                        return !path.endsWith('.py') && !path.endsWith('.txt');
```

Comment:
That's an interesting way to determine if something is a directory :) Maybe use types instead? Instead of using strings, we could have `fileList: (File | Directory)[]` or something like that?

Reply:
Fixed.